### PR TITLE
Deployment preparation: Fraxtal root gauge factory

### PR DESCRIPTION
**Pull request by jubeira**
# Deployment preparation: `FraxtalRootGaugeFactory`

Build-info: https://github.com/balancer/balancer-v2-monorepo/commit/281e98f78dca2c3aabb990c5841e5036c12fa7f6.
The code is very similar to the Optimism Root Gauge; it just adds a network tag to identify the target L2.

The fork test is a mix between 'modern' gauge tests (Avax, Base) which use LZ-based gauges with the Optimism Root Gauge Factory test. The setup is the latest (i.e. it creates a synthetic veBAL whale) and all the cooldowns and periods are fixed, and it checks the Optimism bridge events (which are the same in OP and Fraxtal).

About the inputs: L1 token bridge and Fraxtal BAL addresses naturally differ between OP and Fraxtal. Besides that, the initial gas limit also differs: the one in this task was taken from a test TX.

Deposit TX sample: https://etherscan.io/tx/0xbbff35c8add8d11f2fac562731ec9dab58582c3a14b64580d01b8c757fc976bd
Recipient TX: https://fraxscan.com/tx/0x6b27d68f24275892a22c0a53d2de40cd4610c465b7c1233dc917580c0a855fc9

## Deployment preparation tasks checklist:

- [x] Deployments `README` is updated with the new task name and link
- [x] Task `README` has a description and links to the artifacts
- [x] `build-info` is updated with artifacts compiled at the specified commit
- [x] Artifacts are generated from `build-info`
- [x] New task has a working fork test 

## Code checklist:

- [x] The diff is legible and has no extraneous changes 
- [x] Complex code has been commented in tests and scripts
- [x] Fork test block number and test cases are appropriate for this deployment
- [x] The base branch is either `master`, or there's a description of how to merge

## Issue Resolution

N/A

### CODE REVIEW

File: tasks/20240522-fraxtal-root-gauge-factory/index.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { Task, TaskRunOptions } from '@src';
+import { FraxtalRootGaugeFactoryDeployment } from './input';
+
+export default async (task: Task, { force, from }: TaskRunOptions = {}): Promise<void> => {
+  const input = task.input() as FraxtalRootGaugeFactoryDeployment;
+
+  const args = [input.Vault, input.BalancerMinter, input.L1StandardBridge, input.FraxtalBAL, input.GasLimit];
+
+  const factory = await task.deployAndVerify('OptimismRootGaugeFactory', args, from, force);
```
Comment (EndymionJkb) [inline tasks/20240522-fraxtal-root-gauge-factory/index.ts:9]: Should we call this (and the file) `FraxtalRootGaugeFactory` for consistency with other root gauges? Arbitrum also uses the same build-info, but is called "ArbitrumRootGauge". Even though it works, it looks like a "mistake" (e.g., in the readme, as we have two different tasks pointing to files of the same name).

**Review by EndymionJkb — COMMENTED** [review]
See comment on naming. Also, we usually include manual bridge transactions for root gauges, to ensure the events are the same, etc. (e.g., https://github.com/balancer/balancer-deployments/pull/80)

File: tasks/20240522-fraxtal-root-gauge-factory/index.ts (hunk 1 above)
Comment (jubeira) [reply tasks/20240522-fraxtal-root-gauge-factory/index.ts:9]: Arbitrum has its own gauge; this one has the exact same code as Optimism's gauge.
I tried doing that, but the problem is that the code inside the build info is called `OptimismRootGaugeFactory`. If I name it after Frax, it won't find the contract to deploy.

**Comment by jubeira** [comment]
@EndymionJkb added deposit tx to description (it's also in the inputs script).

File: tasks/20240522-fraxtal-root-gauge-factory/index.ts (hunk 1 above)
Comment (jubeira) [reply tasks/20240522-fraxtal-root-gauge-factory/index.ts:9]: To change the name we'd need to copy the file, rename the contract, and rebuild. I think it's not worth it.

**Review by EndymionJkb — APPROVED** [review]

**Comment by jubeira** [comment]
Updated build info and fork test now with gauge that has network tag. PTAL @EndymionJkb

**Review by EndymionJkb — APPROVED** [review]
I verified that the build-info matches and tests pass. Looks good! Just some minor docs suggestions.

**Review by EndymionJkb — APPROVED** [review]
LGTM

**Pull request merged**